docs_src/security/Ch-Configuring-Add-On-Services.md

```diff
@@ -1 +1,206 @@
+for the service in gating fashion and related Docker volumes from `edgex-init` and secretstore token which
+will be outlined in the next section.
+
+A good example of this will be like `app-serive-rules`:
+
+```yaml
+...
+  app-service-rules:
+    entrypoint: ["/edgex-init/ready_to_run_wait_install.sh"]
+    command: "/app-service-configurable ${DEFAULT_EDGEX_RUN_CMD_PARMS}"
+    volumes:
+      - edgex-init:/edgex-init:ro,z
+      - /tmp/edgex/secrets/app-rules-engine:/tmp/edgex/secrets/app-rules-engine:ro,z
+    depends_on:
+      - security-bootstrapper
+...
+
+```
+
+## Configure the secret store to use
+
+Make sure the TOML configuration file of add-on service like `device-camera` contains
+the proper `[SecretStore]` section.
+
+Example:
+
+```toml
+[SecretStore]
+Type = 'vault'
+Host = 'localhost'
+Port = 8200
+Path = 'device-camera/'
+Protocol = 'http'
+RootCaCertPath = ''
+ServerName = ''
+TokenFile = '/tmp/edgex/secrets/device-camera/secrets-token.json'
+  [SecretStore.Authentication]
+  AuthType = 'X-Vault-Token'
+```
+
+and then in the EdgeX service `secretstore-setup` environment section of `docker-compose` file to add
+the service key or hostname to the environment variable `ADD_SECRETSTORE_TOKENS`:
+
+```yaml
+...
+  secretstore-setup:
+    container_name: edgex-secretstore-setup
+    depends_on:
+    - security-bootstrapper
+    - vault
+    environment:
+      ADD_SECRETSTORE_TOKENS: 'device-camera'
+...
+
+```
+
+With that, `secretstore-setup` then can generate secretstore token from `Vault` and store it in
+the `TokenFile` path specified in the TOML configuration file like the above example.
+
+## Configure the secure Redis message bus to use
+
+The default internal database of EdgeX micro-services is `Redis` and to use Redis message bus,
+we can configure it the EdgeX service `secretstore-setup` environment section of
+`docker-compose` file to add the `redisdb` and the service key using it.
+
+So given an example for service `device-virtual` to use the `Redis` message bus in secure mode,
+we need to tell the `secretstore-setup` to have the service "known" for the secret path of
+the Redis credentials (path for hashed password basically). This can be done through
+the configuration of adding `redisdb[device-virtual]` into the environment variable
+`ADD_KNOWN_SECRETS` in the service `secretstore-setup`, in which `redisdb` is the
+database resource name and the `device-virtual` is the service key or hostname.
+
+```yaml
+...
+  secretstore-setup:
+    container_name: edgex-secretstore-setup
+    depends_on:
+    - security-bootstrapper
+    - vault
+    environment:
+      ADD_SECRETSTORE_TOKENS: 'device-camera, my-service'
+      ADD_KNOWN_SECRETS: redisdb[app-rules-engine],redisdb[device-rest],redisdb[device-virtual]
+...
+
+```
+
+In the above `docker-compose` section of `secretstore-setup`, we specify the known secrets of
+`Redis` database credentials to add to services `app-rules-engine`, `device-rest`,
```

Comment:
Maybe something along the lines of "copies the redisdb secret into device-rest's secret store."

Reply:
modified.